### PR TITLE
add multi-image per gpu support for RCNN

### PR DESCRIPTION
**Pull request by Jerryzcn**
Add multi-image per device support for RCNN.

### CODE REVIEW

File: gluoncv/data/batchify.py [hunk 1]
```diff
@@ -342,10 +372,123 @@ def __call__(self, data):
...
+        in_shapes = [img.shape if len(img.shape) == 2 else img.shape[1:] for shard in sharded_img
+                     for img in shard]
+        sharded_cls_targets, sharded_box_targets, sharded_box_masks = [], [], []
+        for in_shape, cls_targets, box_targets, box_masks in zip(in_shapes, *data[2:]):
+            padded_cls_targets, padded_box_targets, padded_box_masks = [], [], []
+            for feat_sym, cls_target, box_target, box_mask in zip(self._feat_sym, cls_targets,
+                                                                  box_targets, box_masks):
+                _, _, w, h = feat_sym.infer_shape(data=(1, 3, in_shape[0], in_shape[1]))[1][0]
+                padded_cls_target = mx.nd.ones(shape=(w, h, cls_target.shape[-1]),
+                                               ctx=mx.Context('cpu_shared', 0)) * -1.0
```
Comment (zhreshold) [inline gluoncv/data/batchify.py]: not recommanding manipulating array on cpu_shared.

File: gluoncv/data/mscoco/detection.py [hunk 2]
```diff
@@ -104,6 +108,18 @@ def annotation_dir(self):
         """
         return 'annotations'
 
+    def get_im_aspect_ratio(self):
```
Comment (zhreshold) [inline gluoncv/data/mscoco/detection.py]: anyway to get it from coco json? Read raw image is pretty slow

File: gluoncv/data/mscoco/instance.py [hunk 3]
```diff
@@ -85,6 +90,18 @@ def classes(self):
         """Category names."""
         return type(self).CLASSES
 
+    def get_im_aspect_ratio(self):
```
Comment (zhreshold) [inline gluoncv/data/mscoco/instance.py]: same here

File: gluoncv/data/mscoco/detection.py (hunk 2 above)
Comment (Jerryzcn) [reply gluoncv/data/mscoco/detection.py]: PIL only read the image header so it is not that slow, but json could be faster

**Comment by mli** [comment]
Job PR-921-22 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-921/22/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-921-23 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-921/23/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**